src/lambda_function/s3_to_glue/app.py

```diff
@@ -31,38 +73,56 @@ def submit_s3_to_json_workflow(messages: list[dict[str,str]], workflow_name):
     glue_client.put_workflow_run_properties(
         Name=workflow_name,
         RunId=workflow_run["RunId"],
-        RunProperties={"messages": json.dumps(messages)},
+        RunProperties={"messages": json.dumps(objects_info)},
     )
 
 
-def lambda_handler(event, context):
-    """
-    The Lambda entrypoint
-
-    Given a list of files in s3 bucket to process, submit the keys and bucket
-    of the new files to a S3 to JSON Glue workflow.
+def lambda_handler(event, context) -> None:
+    """This main lambda function will be triggered by a
+        SQS event and will poll the SQS queue for
+        all available S3 event messages
 
     Args:
-        event (dict): A fileview object
-        context (obj) : An object that provides methods and properties that
-        provides info about the invocation, function, and execution environment
-
-    Returns:
-        (None) Submits new file records to a Glue workflow.
+        event (json): SQS event
+        context: NA
     """
-    messages = []
-    for record in event["Records"]:
-        bucket_name = record["s3"]["bucket"]["name"]
-        object_key = record["s3"]["object"]["key"]
-        message = {"source_bucket": bucket_name, "source_key": object_key}
-        if "owner.txt" not in object_key:
-            messages.append(message)
-    if len(messages) > 0:
-        logger.info(
+    # Initialize SQS client
+    sqs = boto3.client("sqs")
+    s3_objects_info = []
+    if "Records" in event:
+        for record in event["Records"]:
```

Comment:
What is the scenario where "Records" isn't in the event dict?

Reply:
It logs a message about the event and mentions no messages exist. I haven't encountered a scenario where `Records` wouldn't exist, so it would be interesting to see what that looks like. I wonder if such an event would even have a receipt handle for us to delete it from the queue.

Reply:
Did you want to add a logger?

Reply:
Based on comments below with Phil, it seems like we will just remove the checks for the specific keys (e.g: `Records` in event) in the event JSON because that is the expected format that AWS will provide the SQS message and S3 events in it as and any issues, we'd just get it sent to our DLQ. So probably won't need the logging here.